.github/workflows/build_and_test.yml

```diff
@@ -2,52 +2,98 @@ on: [push, pull_request]
 name: Build and Test check
 jobs:
   rp2040-hal-builds:
-    name: Build rp2040-hal examples without bsp
+    name: Build rp2040-hal's features
     runs-on: ubuntu-20.04
-    strategy:
-      matrix:
-        mode: ["", "--release"]
     steps:
       - uses: actions/checkout@v3
       - uses: dtolnay/rust-toolchain@stable
         with:
           target: thumbv6m-none-eabi
+      - name: Install cargo-hack
+        uses: baptiste0928/cargo-install@v2
+        with:
+          crate: cargo-hack
       - name: Build rp2040-hal's examples
```

Comment:
I *think* this should be called "Build workspace". Examples are built in a different job, below.

Reply:
Good catch, thanks!